Client/Frontend/Browser/TabLocationView.swift

```diff
@@ -430,4 +434,28 @@ private class DisplayTextField: UITextField {
     fileprivate override var canBecomeFirstResponder: Bool {
         return false
     }
+    
+    override func textRect(forBounds bounds: CGRect) -> CGRect {
+        var rect: CGRect = super.textRect(forBounds: bounds)
+        
+        if let size: CGSize = (self.text as NSString?)?.size(withAttributes: [.font: self.font!]) {
+            if size.width > self.bounds.width {
+                rect.origin.x = rect.origin.x - (size.width - self.bounds.width)
+                rect.size.width = rect.size.width + (size.width - self.bounds.width)
+            }
+        }
+        return rect
+    }
+}
+
+extension String {
+    func etldValue(parser: DomainParser?) -> String {
+        guard parser != nil else {
+            return self
```

Comment:
Maybe we should return optional `String?` in this function so we can handle failures better?
And add some logs so we know when the parser fails

Reply:
Covering it in the try-Catch of Parser init. Keeping this as is. :)